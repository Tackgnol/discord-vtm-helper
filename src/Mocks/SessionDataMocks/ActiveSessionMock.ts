import { IActiveSession } from '../../Models/AppModels';
import { existingGTQueryWithoutValue } from '../QueryMocks/globalTestQueries';

export const activeSessionMock: IActiveSession = {
	channelId: '0',
	gameId: '0',
	prevCommand: existingGTQueryWithoutValue,
};
