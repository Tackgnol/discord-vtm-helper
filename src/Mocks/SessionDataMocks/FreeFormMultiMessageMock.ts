export const freeFormMultiMessageMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['2', '3', '4'],
};

export const freeFormMultiMessageErrorMock = {
	reply: 'test',
	userList: ['1', '2', '3'],
	channelMembers: ['4', '5', '6'],
};
