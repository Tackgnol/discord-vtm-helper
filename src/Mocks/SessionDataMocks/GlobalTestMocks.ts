import { GlobalTest } from '../../Models/GameData';

export const globalTestWithoutShortCircuitMock: GlobalTest = {
	testMessage: 'test',
	shortCircuit: false,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'test',
	replyPrefix: 'testPrefix',
};
export const globalTestWithShortCircuitMock: GlobalTest = {
	testMessage: 'shortCircuitTest',
	shortCircuit: true,
	globaltestoptionSet: [
		{ minResult: 1, resultMessage: 'result for 1' },
		{ minResult: 5, resultMessage: 'result for 5' },
	],
	name: 'scTest',
	replyPrefix: 'testPrefix',
};
