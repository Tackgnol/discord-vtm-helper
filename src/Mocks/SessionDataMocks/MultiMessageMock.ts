import { IMultiPlayerMessage } from '../../Models/GameData';

export const multiMessageMock: IMultiPlayerMessage = {
	userList: ['0'],
	name: 'test',
	message: "you've got mail",
};
