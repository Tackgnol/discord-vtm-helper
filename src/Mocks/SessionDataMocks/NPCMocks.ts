import { INPC } from '../../Models/GameData';

export const testNPCWithFacts: INPC = {
	callName: 'testWithFacts',
	description: 'this is a test',
	image: 'placeholder',
	name: 'Test',
	facts: ['fact1', 'fact2'],
};

export const testNPCWithoutFacts: INPC = {
	callName: 'testWithout',
	description: 'this is a test',
	image: 'placeholder',
	name: 'Test',
	facts: [],
};
