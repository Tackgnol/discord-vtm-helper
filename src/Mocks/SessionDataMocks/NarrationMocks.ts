import { INarration } from '../../Models/GameData';

export const basicNarration: INarration = {
	name: 'test',
	image: 'urlToImage',
	narrationText: 'this is a narration text',
};
