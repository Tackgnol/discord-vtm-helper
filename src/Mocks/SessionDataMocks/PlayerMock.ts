import { Player } from '../../Models/GameData';
import { testNPCWithFacts, testNPCWithoutFacts } from './NPCMocks';

export const playerMock: Player = {
	npcSet: [testNPCWithFacts, testNPCWithoutFacts],
	discordUserName: 'test',
	statisticsSet: [{ name: 'test', value: 2 }],
	id: '0',
};
