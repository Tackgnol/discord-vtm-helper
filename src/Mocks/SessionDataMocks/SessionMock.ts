import { SessionData } from '../../Models/AppModels';
import { globalTestWithoutShortCircuitMock, globalTestWithShortCircuitMock } from './GlobalTestMocks';
import { multiMessageMock } from './MultiMessageMock';
import { basicNarration } from './NarrationMocks';
import { statInsightMockHigh, statInsightMockLow } from './StatInsightMocks';

export const sessionMock: SessionData = {
	channelId: '0',
	globaltestSet: [globalTestWithoutShortCircuitMock, globalTestWithShortCircuitMock],
	multiMessageSet: [multiMessageMock],
	narrationSet: [basicNarration],
	statInsightSet: [statInsightMockHigh, statInsightMockLow],
};
