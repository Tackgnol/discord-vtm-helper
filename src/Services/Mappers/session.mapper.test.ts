import chai, { expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';
import { sessionMapper } from './session.mapper';
import { sessionMock } from '../../Mocks/SessionDataMocks/SessionMock';

chai.use(jestSnapshotPlugin());

describe('Services >> Mappers >> sessionMapper', () => {
	it('properly maps a valid session', () => {
		const result = sessionMapper(sessionMock);
		expect(result).toMatchSnapshot();
		expect(result).to.exist;
	});

	it('throws on missing channelId', () => {
		expect(() => {
			sessionMapper({ ...sessionMock, channelId: undefined });
		}).to.throw(Error);
	});
});
