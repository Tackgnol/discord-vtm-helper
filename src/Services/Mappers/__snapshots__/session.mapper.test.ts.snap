// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services >> Mappers >> sessionMapper properly maps a valid session 1`] = `
Object {
  "channelId": "0",
  "globaltestSet": Array [
    Object {
      "globaltestoptionSet": Array [
        Object {
          "minResult": 1,
          "resultMessage": "result for 1",
        },
        Object {
          "minResult": 5,
          "resultMessage": "result for 5",
        },
      ],
      "name": "test",
      "replyPrefix": "testPrefix",
      "shortCircuit": false,
      "testMessage": "test",
    },
    Object {
      "globaltestoptionSet": Array [
        Object {
          "minResult": 1,
          "resultMessage": "result for 1",
        },
        Object {
          "minResult": 5,
          "resultMessage": "result for 5",
        },
      ],
      "name": "scTest",
      "replyPrefix": "testPrefix",
      "shortCircuit": true,
      "testMessage": "shortCircuitTest",
    },
  ],
  "multiMessageSet": Array [
    Object {
      "message": "you've got mail",
      "name": "test",
      "userList": Array [
        "0",
      ],
    },
  ],
  "narrationSet": Array [
    Object {
      "image": "urlToImage",
      "name": "test",
      "narrationText": "this is a narration text",
    },
  ],
  "statInsightSet": Array [
    Object {
      "minValue": 5,
      "name": "test",
      "statName": "test",
      "successMessage": "winner is you!",
    },
    Object {
      "minValue": 1,
      "name": "test",
      "statName": "test",
      "successMessage": "winner is you!",
    },
  ],
}
`;
